src/pkg/services/m365/api/item_serialization_test.go

```diff
@@ -0,0 +1,144 @@
+package api_test
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/alcionai/clues"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/alcionai/corso/src/internal/common/ptr"
+	"github.com/alcionai/corso/src/internal/connector/support"
+	"github.com/alcionai/corso/src/internal/tester"
+	"github.com/alcionai/corso/src/pkg/services/m365/api"
+)
+
+type ItemSerializationUnitSuite struct {
+	tester.Suite
+}
+
+func TestItemSerializationUnitSuite(t *testing.T) {
+	suite.Run(t, &ItemSerializationUnitSuite{Suite: tester.NewUnitSuite(t)})
+}
+
+func (suite *ItemSerializationUnitSuite) TestConcurrentItemSerialization() {
+	var (
+		user      = "a-user"
+		instances = 100
+	)
+
+	table := []struct {
+		name                   string
+		serializer             func(t *testing.T, ctx context.Context, idx int) []byte
+		deserializeAndGetField func(t *testing.T, bs []byte) string
+	}{
+		{
+			name: "Exchange Mail",
+			serializer: func(t *testing.T, ctx context.Context, idx int) []byte {
+				subject := fmt.Sprintf("%d", idx)
+
+				item := models.NewMessage()
+				item.SetSubject(&subject)
+
+				bs, err := api.Mail{}.Serialize(ctx, item, user, subject)
+				require.NoError(t, err, clues.ToCore(err))
+
+				return bs
+			},
+			deserializeAndGetField: func(t *testing.T, bs []byte) string {
+				item, err := support.CreateMessageFromBytes(bs)
+				require.NoError(
+					t,
+					err,
+					"deserializing message of %q: %v",
+					string(bs),
+					clues.ToCore(err))
+
+				return ptr.Val(item.GetSubject())
+			},
+		},
+		{
+			name: "Exchange Event",
+			serializer: func(t *testing.T, ctx context.Context, idx int) []byte {
+				subject := fmt.Sprintf("%d", idx)
+
+				item := models.NewEvent()
+				item.SetSubject(&subject)
+
+				bs, err := api.Events{}.Serialize(ctx, item, user, subject)
+				require.NoError(t, err, clues.ToCore(err))
+
+				return bs
+			},
+			deserializeAndGetField: func(t *testing.T, bs []byte) string {
+				item, err := support.CreateEventFromBytes(bs)
+				require.NoError(
+					t,
+					err,
+					"deserializing event of %q: %v",
+					string(bs),
+					clues.ToCore(err))
+
+				return ptr.Val(item.GetSubject())
+			},
+		},
+		{
+			name: "Exchange Contact",
+			serializer: func(t *testing.T, ctx context.Context, idx int) []byte {
+				name := fmt.Sprintf("%d", idx)
+
+				item := models.NewContact()
+				item.SetGivenName(&name)
+
+				bs, err := api.Contacts{}.Serialize(ctx, item, user, name)
+				require.NoError(t, err, clues.ToCore(err))
+
+				return bs
+			},
+			deserializeAndGetField: func(t *testing.T, bs []byte) string {
+				item, err := support.CreateContactFromBytes(bs)
+				require.NoError(
```

Comment:
iirc the `f` is still required for formatting.
```suggestion
				require.NoErrorf(
```

Reply:
it actually seems to format it properly without the f suffix. I'm not sure if that's some change in the API or not 🤔 without the format specifier in the string it actually complains that there's extra args when printing

Reply:
Oh... that's right.  assert does this ridiculous thing where if it's given `assert.Foo(t, bar, "reason", a, b, c)`, then it'll call `fmt.Printf("reason", a, b, c)` instead of `fmt.Print...`, which would make sense.  So you can pass in a format templated string and vars (`"reason %v %v %v", a, b, c`) without `f` and it'll work.  But if you skip the format templating it'll error.  Which is, yay.